articles/sorting-algorithms.md

```diff
@@ -0,0 +1,145 @@
+# Sorting algorithms visualised
+- 19th December 2016
+
+Sometimes you or a computer needs to sort a list of items—say, a list of names into alphabetical order, or a list of objects into order of length. To sort this list, especially if it is a large list, you need some kind of process: for example, if you were sorting a list of names into alphabetic order, you could put all the names beginning with A together, then all the Bs together, and then you could sort them by the second letter: all the AAs together, then all the ABs together, until they're sorted.
+
+Algorithms used to sort a list of items are called sorting algorithms, and in this article I'm going to explain and visualise a few of the most well known sorting algorithms. Each section will contain one sorting algorithm, a description of the algorithm, and then a visualisation of the algorithm, sorting different length bars (seen below) from shortest to longest, powered by [D3.js]. There are some controls to the bottom right of the screen: it might help if you adjust the speed of the visualisation a bit.
+
+<svg class="sorting-demo" id="unordered" height="300" width="800" viewBox="0 0 800 300"></svg>
+
+Most of the sorting algorithms in this article are comparison sorts: two items are chosen from the list and compared to each other to see which should come first.
+
+----
+
+## Bubble sort
+
+Bubble sort is one of the easier sorting algorithms to understand, as it is one of the simplest. It is also one of the slowest to run (although I won't really be talking about algorithm speed in this article).
+
+It works by iterating through the list from the beginning, comparing the current item to the next item, and moving the longest of the two up the list. When it gets to the end of the list, the current item is marked as sorted and the iteration begins again from the beginning of the array.
+
+The orange bars are the bars being compared, and the darker blue bars are the bars that have finished sorting. Watch to the see the largest unsorted bar being grabbed and "bubbling up" the list:
+
+<svg class="sorting-demo" id="bubble" height="300" width="800" viewBox="0 0 800 300"></svg>
+
+----
+
+## Selection sort
+
+Selection sort works by iterating through the list, finding the smallest unsorted item. Once it has found it, it swaps the item with the left-most unsorted item, marks it as sorted, and starts the search again.
```

Comment:
"through the list and finding the..." (switch comma with and)
"found it it swaps the..." (remove comma)